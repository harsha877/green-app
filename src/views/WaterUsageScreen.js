import React from "react"
import { StyleSheet, Text, View, Linking, Dimensions } from "react-native"
import { useStateValue } from "../views/stateProvider"
import Svg, { Path } from "react-native-svg"

const WaterUsageScreen = () => {
	const [{ user, score }] = useStateValue()
	return (
		<View style={styles.resultScreen}>
			<View style={styles.top}>
				<View style={styles.box}>
					<Svg
						height={200}
						width={Dimensions.get("screen").width}
						viewBox="0 0 1440 320"
						style={styles.topWavy}
					>
						<Path
							fill="#2471A3"
							d="M0,192L60,170.7C120,149,240,107,360,112C480,117,600,171,720,197.3C840,224,960,224,1080,208C1200,192,1320,160,1380,144L1440,128L1440,0L1380,0C1320,0,1200,0,1080,0C960,0,840,0,720,0C600,0,480,0,360,0C240,0,120,0,60,0L0,0Z"
						/>
					</Svg>
				</View>
			</View>
			<Text style={styles.headerText}>Results</Text>
			<Text style={styles.resultText}>
				Water Usage: {score} litres/day
			</Text>
			<Text style={styles.resultText1}>
				Canada Average Water Usage: 2432 litres/day
			</Text>
			<View style={styles.bottom}>
				<View style={styles.box}>
					<Svg
						height={200}
						width={Dimensions.get("screen").width}
						viewBox="0 0 1440 320"
						style={styles.bottomWavy}
					>
						<Path
							fill="#2471A3"
							d="M0,64L40,96C80,128,160,192,240,202.7C320,213,400,171,480,149.3C560,128,640,128,720,154.7C800,181,880,235,960,218.7C1040,203,1120,117,1200,74.7C1280,32,1360,32,1400,32L1440,32L1440,320L1400,320C1360,320,1280,320,1200,320C1120,320,1040,320,960,320C880,320,800,320,720,320C640,320,560,320,480,320C400,320,320,320,240,320C160,320,80,320,40,320L0,320Z"
						/>
					</Svg>
				</View>
			</View>
		</View>
	)
}

export default WaterUsageScreen

const styles = StyleSheet.create({
	resultScreen: {
		flex: 1,
		alignItems: "center",
		justifyContent: "center",
	},
	resultText: {
		fontSize: 25,
		textAlign: "center",
		color: "#000000",
		fontWeight: "500",
	},
	resultText1: {
		fontSize: 25,
		textAlign: "center",
		color: "#000000",
		fontWeight: "500",
		marginTop: 40,
	},
	headerText: {
		color: "#2471A3",
		fontSize: 30,
		margin: 40,
		fontWeight: "500",
		textAlign: "center",
	},
	top: {
		position: "absolute",
		width: Dimensions.get("screen").width,
		top: 0,
	},
	bottom: {
		position: "absolute",
		width: Dimensions.get("screen").width,
		bottom: 0,
	},
	box: {
		backgroundColor: "#2471A3",
		height: 80,
	},
	bottomWavy: {
		position: "absolute",
		bottom: 20,
	},
})
