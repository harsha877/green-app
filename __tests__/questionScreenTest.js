import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import WaterUsageScreen from '../src/views/questionScreen';

import { TestWatcher } from 'jest';

var route = { params: { data: [{ questions: ["tr"] }] } }
test("render test for question screen", () => {
    render(<WaterUsageScreen route={route} navigation={NavigationContainer} />)

})

test("question screen component snapshot", () => {
    let component = render(<WaterUsageScreen route={route} navigation={NavigationContainer} />).toJSON();
    expect(component).toMatchSnapshot();
})

describe("question screen Testing Option component", () => {
    let component;
    beforeEach(() => {
        component = render(<WaterUsageScreen route={route} navigation={NavigationContainer} />);

    })

    test("Testing Styles", () => {
        const props = component.toJSON();

        let Styles = props.children[0].children[0].props.style;

        expect(Styles).toStrictEqual({
            color: "#003f69",
            fontSize: 18,
            margin: 5
        });
    })

    test("Testing Styles text", () => {
        const props = component.toJSON();
        //console.log(props);
        let Styles = props.children[0].children[1].props.style;
        console.log(Styles);
        expect(Styles).toStrictEqual({
            fontSize: 20,
            margin: 5,
            fontWeight: "500",
            textAlign: "center",
        })
    })

    test("Testing Styles option1", () => {
            const props = component.toJSON();

            let Styles = props.children[1].children[0].props.style;

            expect(Styles).toStrictEqual({
                margin: 0,
                padding: 0,
                margin: 30,
                marginBottom: 3,
                borderRadius: 25,
                backgroundColor: "#42bff5",
                fontSize: 18,
                textAlign: "center",
            });
        });
        test("Testing Styles option2", () => {
            const props = component.toJSON();

            let Styles = props.children[1].children[1].props.style;

            expect(Styles).toStrictEqual({
                margin: 0,
                padding: 0,
                margin: 30,
                marginBottom: 3,
                borderRadius: 25,
                backgroundColor: "#42bff5",
                fontSize: 18,
                textAlign: "center",
            });
        });
        test("Testing Styles option3", () => {
            const props = component.toJSON();

            let Styles = props.children[1].children[2].props.style;

            expect(Styles).toStrictEqual({
                margin: 0,
                padding: 0,
                margin: 30,
                marginBottom: 3,
                borderRadius: 25,
                backgroundColor: "#42bff5",
                fontSize: 18,
                textAlign: "center",
            });
        });
        test("Testing Styles option4", () => {
            const props = component.toJSON();

            let Styles = props.children[1].children[3].props.style;

            expect(Styles).toStrictEqual({
                margin: 0,
                padding: 0,
                margin: 30,
                marginBottom: 3,
                borderRadius: 25,
                backgroundColor: "#42bff5",
                fontSize: 18,
                textAlign: "center",
            });
        });
        test("Testing Styles navigation", () => {
            const props = component.toJSON();

            let Styles = props.children[2].props.style;

            expect(Styles).toStrictEqual({
                justifyContent: "space-between",
                flexDirection: "row",
                alignItems: "center",
                justifyContent: "center",
                marginTop: 30,
            });
        });
});