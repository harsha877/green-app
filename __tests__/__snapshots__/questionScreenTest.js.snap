// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`question screen component snapshot 1`] = `
<View>
  <View>
    <Text
      style={
        Object {
          "color": "#003f69",
          "fontSize": 18,
          "margin": 5,
        }
      }
    >
      1
      /
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
          "margin": 5,
          "textAlign": "center",
        }
      }
    />
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#42bff5",
          "borderRadius": 25,
          "fontSize": 18,
          "margin": 30,
          "marginBottom": 3,
          "padding": 0,
          "textAlign": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#1ca3ec",
              "borderColor": "black",
              "borderRadius": 20,
              "margin": 4,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "padding": 5,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#42bff5",
          "borderRadius": 25,
          "fontSize": 18,
          "margin": 30,
          "marginBottom": 3,
          "padding": 0,
          "textAlign": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#1ca3ec",
              "borderColor": "black",
              "borderRadius": 20,
              "margin": 4,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "padding": 5,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#42bff5",
          "borderRadius": 25,
          "fontSize": 18,
          "margin": 30,
          "marginBottom": 3,
          "padding": 0,
          "textAlign": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#1ca3ec",
              "borderColor": "black",
              "borderRadius": 20,
              "margin": 4,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "padding": 5,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#42bff5",
          "borderRadius": 25,
          "fontSize": 18,
          "margin": 30,
          "marginBottom": 3,
          "padding": 0,
          "textAlign": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#1ca3ec",
              "borderColor": "black",
              "borderRadius": 20,
              "margin": 4,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 18,
                "padding": 5,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 30,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#1ca3ec",
            "borderColor": "black",
            "borderRadius": 20,
            "margin": 4,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 18,
              "padding": 5,
              "textAlign": "center",
            }
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</View>
`;
